internal/grpc/server.go

```diff
@@ -17,107 +17,40 @@ package grpc
 import (
 	"context"
 
-	"google.golang.org/grpc"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-
+	api "github.com/envoyproxy/go-control-plane/envoy/api/v2"
+	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
+	"github.com/envoyproxy/go-control-plane/pkg/cache/v2"
+	xds "github.com/envoyproxy/go-control-plane/pkg/server/v2"
 	grpc_prometheus "github.com/grpc-ecosystem/go-grpc-prometheus"
 	"github.com/prometheus/client_golang/prometheus"
-
-	v2 "github.com/envoyproxy/go-control-plane/envoy/api/v2"
-	discovery "github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2"
-	loadstats "github.com/envoyproxy/go-control-plane/envoy/service/load_stats/v2"
-	"github.com/sirupsen/logrus"
+	"google.golang.org/grpc"
 )
 
 // NewAPI returns a *grpc.Server which responds to the Envoy v2 xDS gRPC API.
-func NewAPI(log logrus.FieldLogger, resources map[string]Resource, registry *prometheus.Registry, opts ...grpc.ServerOption) *grpc.Server {
+func NewAPI(registry *prometheus.Registry, snapshotCache cache.SnapshotCache, opts ...grpc.ServerOption) *grpc.Server {
 	s := &grpcServer{
```

Comment:
Seems like `s` and the `grpcServer` struct def can be removed since they're not being used, and the metrics can be handled separately.